src/Microsoft.DotNet.Watcher.Tools/CommandLineOptions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    internal class CommandLineOptions
+    {
+        public bool IsHelp { get; set; }
+        public IList<string> RemainingArguments { get; set; }
+        public static CommandLineOptions Parse(string[] args, TextWriter consoleOutput)
+        {
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            var app = new CommandLineApplication(throwOnUnexpectedArg: false)
+            {
+                Name = "dotnet watch",
+                FullName = "Microsoft DotNet File Watcher",
```

Comment:
Yes. It looks a little weird, but it's what the [DotNet team uses](https://github.com/dotnet/cli/tree/rel/1.0.0/src).